### ISSUE TITLE
tempo doesn't avoid beams

### ISSUE THREAD

**Issue by craigsapp** [rism-ch/verovio#418]
tempo floating elements avoid noteheads, but not beams:

Here is a case where the tempo element overlaps a beam:

<img width="307" alt="screen shot 2017-01-08 at 23 27 09" src="https://cloud.githubusercontent.com/assets/3487289/21759584/34e8e314-d5fa-11e6-984c-0a2d1e6b1431.png">

MEI data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt>
                <date>2017-01-08 23:27:01</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Transcoded from Humdrum with Verovio version 0.9.13-dev-8c6bcc9</p>
            </projectDesc>
        </encodingDesc>
        <extMeta>
            <frames xmlns:humxml="http://www.humdrum.org/ns/humxml">
                <metaFrame n="0" token="!!!OMD: Allegro, ma non troppo" xml:id="loc0">
                    <frameInfo>
                        <startTime float="0" />
                        <frameType>reference</frameType>
                        <referenceKey>OMD</referenceKey>
                        <referenceValue>Allegro, ma non troppo</referenceValue>
                    </frameInfo>
                </metaFrame>
            </frames>
        </extMeta>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-000000062449016" midi.bpm="132">
                        <staffGrp xml:id="m-000000201317582" symbol="brace" barthru="true">
                            <staffDef xml:id="staffdef-000000033374266" clef.shape="G" clef.line="2" key.sig="1f" meter.count="3" meter.unit="4" n="1" lines="5" />
                            <staffDef xml:id="staffdef-000000084982203" clef.shape="F" clef.line="4" key.sig="1f" meter.count="3" meter.unit="4" n="2" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-000000071356959">
                        <measure xml:id="measure-L9" n="1">
                            <staff xml:id="staff-L9F2" n="1">
                                <layer xml:id="layer-L9F2" n="1">
                                    <beam xml:id="beam-000000153001482">
                                        <chord xml:id="chord-L11F2" dots="1" dur="8">
                                            <note xml:id="note-L11F2S1" oct="4" pname="a" accid.ges="n" />
                                            <note xml:id="note-L11F2S2" oct="5" pname="c" accid.ges="n" />
                                        </chord>
                                        <chord xml:id="chord-L12F2" dur="16">
                                            <note xml:id="note-L12F2S1" oct="4" pname="a" accid.ges="n" />
                                            <note xml:id="note-L12F2S2" oct="5" pname="c" accid.ges="n" />
                                        </chord>
                                    </beam>
                                    <chord xml:id="chord-L13F2" dur="4">
                                        <note xml:id="note-L13F2S1" oct="4" pname="a" accid.ges="n" />
                                        <note xml:id="note-L13F2S2" oct="5" pname="c" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L14F2" dur="4">
                                        <note xml:id="note-L14F2S1" oct="4" pname="a" accid.ges="n" />
                                        <note xml:id="note-L14F2S2" oct="5" pname="f" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <staff xml:id="staff-L9F1" n="2">
                                <layer xml:id="layer-L9F1" n="1">
                                    <beam xml:id="beam-000000041628860">
                                        <chord xml:id="chord-L11F1" dots="1" dur="8">
                                            <note xml:id="note-L11F1S1" oct="3" pname="f" accid.ges="n" />
                                            <note xml:id="note-L11F1S2" oct="4" pname="c" accid.ges="n" />
                                        </chord>
                                        <chord xml:id="chord-L12F1" dur="16">
                                            <note xml:id="note-L12F1S1" oct="3" pname="f" accid.ges="n" />
                                            <note xml:id="note-L12F1S2" oct="4" pname="c" accid.ges="n" />
                                        </chord>
                                    </beam>
                                    <chord xml:id="chord-L13F1" dur="4">
                                        <note xml:id="note-L13F1S1" oct="3" pname="f" accid.ges="n" />
                                        <note xml:id="note-L13F1S2" oct="4" pname="c" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L14F1" dur="4">
                                        <note xml:id="note-L14F1S1" oct="3" pname="f" accid.ges="n" />
                                        <note xml:id="note-L14F1S2" oct="4" pname="c" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <tempo xml:id="tempo-000000107674175" staff="1" tstamp="1.000000">Allegro, ma non troppo</tempo>
                            <dynam xml:id="dynam-000000178271215" staff="1" tstamp="1.000000">f</dynam>
                            <pedal xml:id="pedal-000000155720098" staff="2" tstamp="1.000000" dir="down" />
                            <pedal xml:id="pedal-000000117159944" staff="2" tstamp="3.500000" dir="up" />
                        </measure>
                        <measure xml:id="measure-L17" n="2">
                            <staff xml:id="staff-L17F2" n="1">
                                <layer xml:id="layer-L17F2" n="1">
                                    <beam xml:id="beam-000000112303512">
                                        <chord xml:id="chord-L19F3" dots="1" dur="8">
                                            <note xml:id="note-L19F3S1" oct="4" pname="g" accid.ges="n" />
                                            <note xml:id="note-L19F3S2" oct="5" pname="e" accid.ges="n" />
                                        </chord>
                                        <chord xml:id="chord-L20F3" dur="16">
                                            <note xml:id="note-L20F3S1" oct="4" pname="f" accid.ges="n" />
                                            <note xml:id="note-L20F3S2" oct="5" pname="d" accid.ges="n" />
                                        </chord>
                                    </beam>
                                    <chord xml:id="chord-L21F3" dur="2">
                                        <note xml:id="note-L21F3S1" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L21F3S2" oct="5" pname="c" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <staff xml:id="staff-L17F1" n="2">
                                <layer xml:id="layer-L17F1" n="1">
                                    <beam xml:id="beam-000000184609385">
                                        <chord xml:id="chord-L19F1" dots="1" dur="8">
                                            <note xml:id="note-L19F1S1" oct="3" pname="g" accid.ges="n" />
                                            <note xml:id="note-L19F1S2" oct="4" pname="c" accid.ges="n" />
                                        </chord>
                                        <note xml:id="note-L20F1" dur="16" accid="n" oct="3" pname="b" />
                                    </beam>
                                    <note xml:id="note-L21F1" dur="2" oct="4" pname="c" accid.ges="n" />
                                </layer>
                                <layer xml:id="layer-L19F2" n="2">
                                    <note xml:id="note-L19F2" dots="1" dur="2" oct="3" pname="c" accid.ges="n" />
                                </layer>
                            </staff>
                            <slur xml:id="slur-000000189708348" staff="1" startid="#chord-L19F3" endid="#chord-L21F3" />
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

If there are noteheads in the way instead, the tempo avoids them:

<img width="321" alt="screen shot 2017-01-08 at 23 30 33" src="https://cloud.githubusercontent.com/assets/3487289/21759619/77e4898e-d5fa-11e6-857e-1da1c7e24779.png">

MEI data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt>
                <date>2017-01-08 23:31:01</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Transcoded from Humdrum with Verovio version 0.9.13-dev-8c6bcc9</p>
            </projectDesc>
        </encodingDesc>
        <extMeta>
            <frames xmlns:humxml="http://www.humdrum.org/ns/humxml">
                <metaFrame n="0" token="!!!OMD: Allegro, ma non troppo" xml:id="loc0">
                    <frameInfo>
                        <startTime float="0" />
                        <frameType>reference</frameType>
                        <referenceKey>OMD</referenceKey>
                        <referenceValue>Allegro, ma non troppo</referenceValue>
                    </frameInfo>
                </metaFrame>
                <metaFrame n="25" token="!!!Xfilter: transpose -b 40" xml:id="loc25">
                    <frameInfo>
                        <startTime float="6" />
                        <frameType>reference</frameType>
                        <referenceKey>Xfilter</referenceKey>
                        <referenceValue>transpose -b 40</referenceValue>
                    </frameInfo>
                </metaFrame>
            </frames>
        </extMeta>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-000000174560541" midi.bpm="132">
                        <staffGrp xml:id="m-000000152850237" symbol="brace" barthru="true">
                            <staffDef xml:id="staffdef-000000057635904" clef.shape="G" clef.line="2" key.sig="1f" meter.count="3" meter.unit="4" n="1" lines="5" />
                            <staffDef xml:id="staffdef-000000181107245" clef.shape="F" clef.line="4" key.sig="1f" meter.count="3" meter.unit="4" n="2" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-000000025401419">
                        <measure xml:id="measure-L10" n="1">
                            <staff xml:id="staff-L10F2" n="1">
                                <layer xml:id="layer-L10F2" n="1">
                                    <beam xml:id="beam-000000137281608">
                                        <chord xml:id="chord-L12F2" dots="1" dur="8">
                                            <note xml:id="note-L12F2S1" oct="5" pname="a" accid.ges="n" />
                                            <note xml:id="note-L12F2S2" oct="6" pname="c" accid.ges="n" />
                                        </chord>
                                        <chord xml:id="chord-L13F2" dur="16">
                                            <note xml:id="note-L13F2S1" oct="5" pname="a" accid.ges="n" />
                                            <note xml:id="note-L13F2S2" oct="6" pname="c" accid.ges="n" />
                                        </chord>
                                    </beam>
                                    <chord xml:id="chord-L14F2" dur="4">
                                        <note xml:id="note-L14F2S1" oct="5" pname="a" accid.ges="n" />
                                        <note xml:id="note-L14F2S2" oct="6" pname="c" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L15F2" dur="4">
                                        <note xml:id="note-L15F2S1" oct="5" pname="a" accid.ges="n" />
                                        <note xml:id="note-L15F2S2" oct="6" pname="f" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <staff xml:id="staff-L10F1" n="2">
                                <layer xml:id="layer-L10F1" n="1">
                                    <beam xml:id="beam-000000167706977">
                                        <chord xml:id="chord-L12F1" dots="1" dur="8">
                                            <note xml:id="note-L12F1S1" oct="4" pname="f" accid.ges="n" />
                                            <note xml:id="note-L12F1S2" oct="5" pname="c" accid.ges="n" />
                                        </chord>
                                        <chord xml:id="chord-L13F1" dur="16">
                                            <note xml:id="note-L13F1S1" oct="4" pname="f" accid.ges="n" />
                                            <note xml:id="note-L13F1S2" oct="5" pname="c" accid.ges="n" />
                                        </chord>
                                    </beam>
                                    <chord xml:id="chord-L14F1" dur="4">
                                        <note xml:id="note-L14F1S1" oct="4" pname="f" accid.ges="n" />
                                        <note xml:id="note-L14F1S2" oct="5" pname="c" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L15F1" dur="4">
                                        <note xml:id="note-L15F1S1" oct="4" pname="f" accid.ges="n" />
                                        <note xml:id="note-L15F1S2" oct="5" pname="c" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <tempo xml:id="tempo-000000044822516" staff="1" tstamp="1.000000">Allegro, ma non troppo</tempo>
                            <dynam xml:id="dynam-000000109408673" staff="1" tstamp="1.000000">f</dynam>
                            <pedal xml:id="pedal-000000043416078" staff="2" tstamp="1.000000" dir="down" />
                            <pedal xml:id="pedal-000000045713399" staff="2" tstamp="3.500000" dir="up" />
                        </measure>
                        <measure xml:id="measure-L18" n="2">
                            <staff xml:id="staff-L18F2" n="1">
                                <layer xml:id="layer-L18F2" n="1">
                                    <beam xml:id="beam-000000023968634">
                                        <chord xml:id="chord-L20F3" dots="1" dur="8">
                                            <note xml:id="note-L20F3S1" oct="5" pname="g" accid.ges="n" />
                                            <note xml:id="note-L20F3S2" oct="6" pname="e" accid.ges="n" />
                                        </chord>
                                        <chord xml:id="chord-L21F3" dur="16">
                                            <note xml:id="note-L21F3S1" oct="5" pname="f" accid.ges="n" />
                                            <note xml:id="note-L21F3S2" oct="6" pname="d" accid.ges="n" />
                                        </chord>
                                    </beam>
                                    <chord xml:id="chord-L22F3" dur="2">
                                        <note xml:id="note-L22F3S1" oct="5" pname="e" accid.ges="n" />
                                        <note xml:id="note-L22F3S2" oct="6" pname="c" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <staff xml:id="staff-L18F1" n="2">
                                <layer xml:id="layer-L18F1" n="1">
                                    <beam xml:id="beam-000000086498569">
                                        <chord xml:id="chord-L20F1" dots="1" dur="8">
                                            <note xml:id="note-L20F1S1" oct="4" pname="g" accid.ges="n" />
                                            <note xml:id="note-L20F1S2" oct="5" pname="c" accid.ges="n" />
                                        </chord>
                                        <note xml:id="note-L21F1" dur="16" accid="n" oct="4" pname="b" />
                                    </beam>
                                    <note xml:id="note-L22F1" dur="2" oct="5" pname="c" accid.ges="n" />
                                </layer>
                                <layer xml:id="layer-L20F2" n="2">
                                    <note xml:id="note-L20F2" dots="1" dur="2" oct="4" pname="c" accid.ges="n" />
                                </layer>
                            </staff>
                            <slur xml:id="slur-000000002680650" staff="1" startid="#chord-L20F3" endid="#chord-L22F3" />
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by lpugin** [comment]
It seems that it relates to the stem direction problem with beamed chords (#414). With`stem.dir="up"` on the chords within the beams that it works properly.

**Comment by lpugin** [comment]
Fixed with 7de376debeaa69323358c03f053b1b24595ab0b2

**Comment by craigsapp** [comment]
Looks good:

<img width="732" alt="screen shot 2017-03-06 at 2 45 42 pm" src="https://cloud.githubusercontent.com/assets/3487289/23633731/ca2e760c-027b-11e7-9722-358d8c8e22ee.png">

<img width="732" alt="screen shot 2017-03-06 at 2 46 53 pm" src="https://cloud.githubusercontent.com/assets/3487289/23633733/ce34cec2-027b-11e7-9284-5e962c51d9a8.png">
